Library/Homebrew/cmd/update.sh

```diff
@@ -754,15 +754,28 @@ EOS
   if [[ -n "${HOMEBREW_INSTALL_FROM_API}" ]]
   then
     mkdir -p "${HOMEBREW_CACHE}/api"
+    if [[ -f "${HOMEBREW_CACHE}/api/formula.json" ]]
+    then
+      INITIAL_JSON_BYTESIZE="$(wc -c "${HOMEBREW_CACHE}"/api/formula.json)"
+    fi
     curl \
       "${CURL_DISABLE_CURLRC_ARGS[@]}" \
       --fail --compressed --silent --max-time 5 \
       --location --remote-time --output "${HOMEBREW_CACHE}/api/formula.json" \
       --time-cond "${HOMEBREW_CACHE}/api/formula.json" \
       --user-agent "${HOMEBREW_USER_AGENT_CURL}" \
       "https://formulae.brew.sh/api/formula.json"
-    # TODO: we probably want to print an error if this fails.
-    # TODO: set HOMEBREW_UPDATED or HOMEBREW_UPDATE_FAILED
+    curl_exit_code=$?
+    if [[ ${curl_exit_code} -eq 0 ]]
+    then
+      CURRENT_JSON_BYTESIZE="$(wc -c "${HOMEBREW_CACHE}"/api/formula.json)"
+      if [[ "${INITIAL_JSON_BYTESIZE}" != "${CURRENT_JSON_BYTESIZE}" ]]
+      then
+        HOMEBREW_UPDATED="1"
+      fi
```

Comment:
Thanks for this PR, @hyuraku! Looks great!

Sorry for the late comment. Would it make more sense to do this comparison based on the file update time? We pass `--remote-time` to `curl` and use this modification time anyway with `--time-cond`, so it seems reasonable to use that to see whether the file was updated. This removes the (unlikely) edge case where the pre and post files are the same thing

Reply:
@Rylan12 @hyuraku If we can do this easily: sure!